src/main/src/main/java/org/geoserver/data/InternationalContentHelper.java

```diff
@@ -0,0 +1,352 @@
+        for (ResourceInfo resourceInfo : resourceInfos) {
+            candidates.addAll(
+                    getLocalesFromInternationalString(resourceInfo.getInternationalTitle()));
+            candidates.addAll(
+                    getLocalesFromInternationalString(resourceInfo.getInternationalAbstract()));
+        }
+        this.supportedLocales = candidates;
+    }
+
+    private void setSupportedLocales(
+            ServiceInfo info, List<LayerInfo> layerInfos, List<LayerGroupInfo> groups) {
+        Set<Locale> candidates = new HashSet<>();
+        candidates.addAll(getLocalesFromInternationalString(info.getInternationalTitle()));
+        candidates.addAll(getLocalesFromInternationalString(info.getInternationalAbstract()));
+        setSupportedLocalesFromGroups(candidates, groups);
+        setSupportedLocalesFromLayers(candidates, layerInfos);
+        this.supportedLocales = candidates;
+    }
+
+    private void setSupportedLocalesFromGroups(
+            Set<Locale> candidates, List<LayerGroupInfo> groups) {
+        for (LayerGroupInfo groupInfo : groups) {
+            candidates.addAll(getLocalesFromInternationalString(groupInfo.getInternationalTitle()));
+            candidates.addAll(
+                    getLocalesFromInternationalString(groupInfo.getInternationalAbstract()));
+            List<StyleInfo> styles = new ArrayList<>(groupInfo.getStyles());
+            setSupportedLocalesFromStyles(candidates, styles);
+        }
+    }
+
+    private void setSupportedLocalesFromLayers(Set<Locale> candidates, List<LayerInfo> layers) {
+        for (LayerInfo layerInfo : layers) {
+            candidates.addAll(getLocalesFromInternationalString(layerInfo.getInternationalTitle()));
+            candidates.addAll(
+                    getLocalesFromInternationalString(layerInfo.getInternationalAbstract()));
+            List<StyleInfo> styles = new ArrayList<>(layerInfo.getStyles());
+            styles.add(layerInfo.getDefaultStyle());
+            setSupportedLocalesFromStyles(candidates, styles);
+        }
+    }
+
+    private void setSupportedLocalesFromStyles(Set<Locale> candidates, List<StyleInfo> styles) {
+        for (StyleInfo styleInfo : styles) {
+            if (styleInfo != null) {
+                try {
+                    Description description = styleInfo.getStyle().getDescription();
+                    if (description != null) {
+                        candidates.addAll(
+                                getLocalesFromInternationalString(description.getTitle()));
+                        candidates.addAll(
+                                getLocalesFromInternationalString(description.getAbstract()));
+                    }
+                } catch (IOException e) {
+                    throw new RuntimeException(e);
+                }
+            }
+        }
+    }
+
+    private Set<Locale> getLocalesFromInternationalString(InternationalString internationalString) {
```

Comment:
The title is descriptive, but very long. Since the internationas string is the only argument, the should be just as readable, but shorter:

```suggestion
    private Set<Locale> getLocales(InternationalString internationalString) {
```

There are other cases in this class that could be simplified accordingly.

Reply:
fixed